docs/usage/shoot_pod_autoscaling_best_practices.md

```diff
@@ -0,0 +1,337 @@
+- **VPA Recommender:** The Recommender observes the historic and current resource metrics of pods and generates recommendations based on this data.
+- **VPA Updater:** The Updater component checks the recommendations from the Recommender and decides whether any pod's resource requests need to be updated. If an update is needed, the Updater will evict the pod.
+- **VPA Admission Controller:** When a pod is (re-)created, the Admission Controller modifies the pod's resource requests based on the recommendations from the Recommender. This ensures that the pod starts with the optimal amount of resources.
+
+## Defining a VPA Resource
+
+To configure VPA, you need to create a VPA resource in your cluster. This resource specifies the target to scale, the metrics to be used for scaling decisions, and the policies for resource updates. Here's an example of an VPA configuration:
+
+```yaml
+apiVersion: autoscaling.k8s.io/v1
+kind: VerticalPodAutoscaler
+metadata:
+  name: foo-vpa
+spec:
+  targetRef:
+    apiVersion: "apps/v1"
+    kind:       Deployment
+    name:       foo-deployment
+  updatePolicy:
+    updateMode: "Auto"
+  resourcePolicy:
+    containerPolicies:
+    - containerName: foo-container
+      controlledValues: RequestsOnly
+      minAllowed:
+        cpu: 50m
+        memory: 200M
+      maxAllowed:
+        cpu: 4
+        memory: 16G
+```
+
+In this example, VPA is configured to scale `foo-deployment` requests (`RequestsOnly`) from 50m cores (`minAllowed`) up to 4 cores (`maxAllowed`) and 200M memory (`minAllowed`) up to 16G memory (`maxAllowed`) automatically (`updateMode`). VPA doesn't support in-place updates, so in `updateMode` `Auto` it will evict pods under certain conditions and then mutate the requests (and possibly limits if you omit `controlledValues` or set it to `RequestsAndLimits`, which is the default) of upcoming new pods.
+
+[Multiple update modes exist](https://github.com/kubernetes/autoscaler/blob/master/vertical-pod-autoscaler/README.md#quick-start). They influence eviction and mutation. The most important ones are:
+- `Off`: In this mode, recommendations are computed, but never applied. This mode is useful, if you want to learn more about your workload or if you have a custom controller that depends on VPA's recommendations but shall act instead of VPA.
+- `Initial`: In this mode, recommendations are computed and applied, but pods are never proactively evicted to enforce new recommendations over time. This mode is useful, if your workload is sensitive to evictions, e.g. some brownfield singleton application or a daemon set pod that is critical for the node.
```

Comment:
Or if you want to have the same behavior as for StatefulSet `updateStrategy: onDelete`

Reply:
Indeed, good point, I wrote now:

> This mode is useful, if you want to control pod evictions yourself (similar to the [`StatefulSet` `updateStrategy` `OnDelete`](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/#update-strategies)) or your workload is sensitive to evictions, e.g. some brownfield singleton application or a daemon set pod that is critical for the node.

Thank you.